metaflow/plugins/kubernetes/kubernetes_cli.py

```diff
@@ -166,6 +174,9 @@ def echo(msg, stream="stderr", job_id=None):
     stdout_location = ds.get_log_location(TASK_LOG_SOURCE, "stdout")
     stderr_location = ds.get_log_location(TASK_LOG_SOURCE, "stderr")
 
+    # `node_selector` is a tuple of strings, convert it to a dictionary
+    node_selector = KubernetesDecorator.parse_node_selector(node_selector)
```

Comment:
is this needed anymore?

Reply:
Yes, it is. At this stage, `node_selector` is a tuple of strings. `kubernetes.launch_job` expects a dictionary

Reply:
Any reason not to handle this parsing within `kubernetes_job` - the actual format is dictated by the kubernetes SDK and that's why currently all the Kubernetes-related formatting is happening within the KubernetesJob object. As the SDK evolves, any changes would be isolated to that object.

Reply:
`node_selector` contains the value generated by
```python
@click.option(
    "--node-selector",
    multiple=True,
    default=None,
    help="NodeSelector for Kubernetes pod.",
)
```
which is a tuple of strings
```python
('key'='val','foo=bar')
```

`kubernetes_job` expect a dictionary like
```python
{
  "key": "val",
  "foo": "bar",
}
```
`parse_node_selector` converts the tuple of strings to a dictionary compatible with the Kubernetes SDK.
Does it make sense?